### PR TITLE
Graph 360 tests

### PR DESCRIPTION
**Pull request by soleinik-figment**

### CODE REVIEW

File: chain/common/tests/resources/acme.proto [hunk 1]
```diff
@@ -0,0 +1,63 @@
+syntax = "proto3";
+
+package sf.acme.type.v1;
+
+option go_package = "github.com/streamingfast/firehose-acme/pb/sf/acme/type/v1;pbacme";
+
+import "firehose/firehose.proto";
+
```
Comment (pienkowb) [inline chain/common/tests/resources/acme.proto:8]: Let's format the file using `clang-format` ([command](https://github.com/figment-networks/proto-cosmos/blob/main/Makefile#L13-L14=), [config](https://github.com/figment-networks/proto-cosmos/blob/main/.clang-format)) to remove such empty lines.

File: graph/proto/firehose.proto [hunk 2]
```diff
@@ -6,6 +6,11 @@ import "google/protobuf/any.proto";
 
 option go_package = "github.com/streamingfast/pbgo/sf/firehose/v1;pbfirehose";
 
+import "google/protobuf/descriptor.proto";
+
+
+
+
```
Comment (pienkowb) [inline graph/proto/firehose.proto:13]: If we don't use this file anymore, we can bring back the original version.

File: chain/common/tests/resources/acme.proto [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  oneof sum {
+    Attribute attribute = 2;
+    BigInt big_int = 3;
+  }
+}
+
+message RequiredTest {
+  string key = 1 [(firehose.required) = true];
+  string value = 2;
+}
```
[the remark this answers is not in the document]
Comment (soleinik-figment) [reply chain/common/tests/resources/acme.proto:63]: orig messages used for volume and there was no enumerations.

File: chain/common/tests/test-acme.rs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+fn check_type_count_ok() {
+    let types = parse_proto_file(PROTO_FILE).expect("Unable to read proto file!");
+    assert_eq!(8, types.len());
+}
+
+#[test]
+fn required_ok() {
+    let types = parse_proto_file(PROTO_FILE).expect("Unable to read proto file!");
+    let msg = types.get("RequiredTest");
+    assert!(msg.is_some(), "\"RequiredTest\" type is not available!");
```
Comment (pienkowb) [inline chain/common/tests/test-acme.rs:15]: Could we test the `Transaction` message instead (which also has the `required` annotation) and get rid of `RequiredTest`?

File: chain/cosmos/proto/type.proto [hunk 5]
```diff
@@ -66,14 +75,15 @@ message Consensus {
...
 
 message Timestamp {
-  int64 seconds = 1; //8
-  int32 nanos = 2;   //4
+  int64 seconds = 1; // 8
+  int32 nanos = 2;   // 4
 }
 
-//message BlockID {
+// message BlockID {
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:82]: I think this line can be removed.

File: chain/cosmos/proto/type.proto [hunk 6]
```diff
@@ -143,26 +175,39 @@ message SignedHeader {
...
-  BLOCK_ID_FLAG_ABSENT = 1 [(gogoproto.enumvalue_customname) = "BlockIDFlagAbsent"];
-  BLOCK_ID_FLAG_COMMIT = 2 [(gogoproto.enumvalue_customname) = "BlockIDFlagCommit"];
-  BLOCK_ID_FLAG_NIL = 3 [(gogoproto.enumvalue_customname) = "BlockIDFlagNil"];
+  BLOCK_ID_FLAG_UNKNOWN = 0
+      [ (gogoproto.enumvalue_customname) = "BlockIDFlagUnknown" ];
+  BLOCK_ID_FLAG_ABSENT = 1
+      [ (gogoproto.enumvalue_customname) = "BlockIDFlagAbsent" ];
+  BLOCK_ID_FLAG_COMMIT = 2
+      [ (gogoproto.enumvalue_customname) = "BlockIDFlagCommit" ];
+  BLOCK_ID_FLAG_NIL = 3 [ (gogoproto.enumvalue_customname) = "BlockIDFlagNil" ];
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:210]: Any reason the formatting here has changed (if it's the same `clang-format` config)?

File: chain/cosmos/proto/type.proto [hunk 7]
```diff
@@ -298,10 +357,7 @@ message ModeInfo {
   }
 }
 
-
-message ModeInfoSingle {
-  SignMode mode = 1;
-}
+message ModeInfoSingle { SignMode mode = 1; }
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:360]: It looks like the default formatting rules were used (instead of [our custom ones](https://github.com/figment-networks/proto-cosmos/blob/main/.clang-format)).

File: chain/cosmos/proto/type.proto [hunk 8]
```diff
@@ -8,11 +8,13 @@ import "google/protobuf/descriptor.proto";
...
 import "gogoproto/gogo.proto";
 import "cosmos_proto/cosmos.proto";
-import "firehose/firehose.proto";
+import "firehose/annotations.proto";
 
 message Block {
-  Header header = 1 [(gogoproto.nullable) = false, (firehose.required) = true];
-  EvidenceList evidence = 2 [(gogoproto.nullable) = false, (firehose.required) = true];
+  Header header = 1
+      [ (gogoproto.nullable) = false, (firehose.required) = true ];
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:15]: I would seriously consider removing the `gogoproto.nullable` annotations from that file but this is outside of the scope of this task. We can tackle that in a separate ticket.

File: chain/cosmos/proto/type.proto [hunk 9]
```diff
@@ -8,11 +8,13 @@ import "google/protobuf/descriptor.proto";
...
+  EvidenceList evidence = 2
+      [ (gogoproto.nullable) = false, (firehose.required) = true ];
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:17]: I would add the `firehose.required` annotation only in the places where the `asc_new_or_missing` function was used in [the `abi.rs` file](https://github.com/figment-networks/graph-node/blob/master/chain/cosmos/src/runtime/abi.rs). 

That is:
* `Block#header`
* `Block#result_begin_block`
* `Block#result_end_block`
* `HeaderOnlyBlock#header`
* `EventData#event`
* `EventData#block`
* `TransactionData#tx`
* `TransactionData#block`
* `Header#last_block_id`
* `TxResult#tx`
* `TxResult#result`
* `Tx#body`

File: chain/cosmos/proto/type.proto [hunk 10]
```diff
@@ -8,14 +8,14 @@ import "google/protobuf/descriptor.proto";
 import "google/protobuf/any.proto";
 import "gogoproto/gogo.proto";
 import "cosmos_proto/cosmos.proto";
-import "firehose/firehose.proto";
+import "firehose/annotations.proto";
 
 message Block {
-  Header header = 1 [(gogoproto.nullable) = false, (firehose.required) = true];
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:14]: Removing `gogoproto.nullable` is the way forward, but this change has to be introduced in `proto-cosmos` first (and we need to test if the whole stack still works).

Let's keep these annotations for now, do the minimal change to keep the same functionality (use `firehose.required` only where `asc_nil_or_missing` was used) and address `gogoproto.nullable` in the future. I've created [a separate ticket](https://figmentio.atlassian.net/browse/GRAPH-477) for that.

File: chain/cosmos/proto/type.proto [hunk 11]
```diff
@@ -8,14 +8,14 @@ import "google/protobuf/descriptor.proto";
...
-  EvidenceList evidence = 2 [(gogoproto.nullable) = false, (firehose.required) = true];
+  Header header = 1 [(firehose.required) = true];
+  EvidenceList evidence = 2 [(firehose.required) = true];
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:15]: We don't need `firehose.required` here.

File: chain/cosmos/proto/type.proto [hunk 12]
```diff
@@ -27,25 +27,25 @@ message Block {
...
 message TransactionData {
-  TxResult tx = 1;
-  HeaderOnlyBlock block = 2;
+  TxResult tx = 1 [(firehose.required) = true];
+  HeaderOnlyBlock block = 2 [(firehose.required) = true];
 }
 
 message Header {
-  Consensus version = 1 [(gogoproto.nullable) = false, (firehose.required) = true];
+  Consensus version = 1 [(firehose.required) = true];
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:44]: We don't need `firehose.required` here.

File: chain/cosmos/proto/type.proto [hunk 13]
```diff
@@ -27,25 +27,25 @@ message Block {
...
   string chain_id = 2 [(gogoproto.customname) = "ChainID"];
   uint64 height = 3;
-  Timestamp time = 4 [(gogoproto.nullable) = false, (firehose.required) = true, (gogoproto.stdtime) = true];
-  BlockId last_block_id = 5 [(gogoproto.nullable) = false, (firehose.required) = true];
+  Timestamp time = 4 [(firehose.required) = true, (gogoproto.stdtime) = true];
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:47]: We don't need `firehose.required` here.

File: chain/cosmos/proto/type.proto [hunk 14]
```diff
@@ -60,20 +60,18 @@ message Header {
 
 message Consensus {
   option (gogoproto.equal) = true;
-
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:63]: Different formatting? Let's make sure we all use the same `clang-format` config: https://github.com/figment-networks/proto-cosmos/blob/main/.clang-format

File: chain/cosmos/proto/type.proto [hunk 15]
```diff
@@ -60,20 +60,18 @@ message Header {
...
-  int32 nanos = 2;   //4
+  int64 seconds = 1;  // 8
+  int32 nanos = 2;    // 4
 }
 
-//message BlockID {
 message BlockId {
   bytes hash = 1;
-  PartSetHeader part_set_header = 2 [(gogoproto.nullable) = false, (firehose.required) = true];
+  PartSetHeader part_set_header = 2 [(firehose.required) = true];
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:74]: We don't need `firehose.required` here.

File: chain/cosmos/proto/type.proto [hunk 16]
```diff
@@ -82,7 +80,7 @@ message PartSetHeader {
 }
 
 message EvidenceList {
-  repeated Evidence evidence = 1 [(gogoproto.nullable) = false, (firehose.required) = true];
+  repeated Evidence evidence = 1 [(firehose.required) = true];
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:83]: We don't need `firehose.required` here.

File: chain/cosmos/proto/type.proto [hunk 17]
```diff
@@ -97,16 +95,20 @@ message DuplicateVoteEvidence {
   EventVote vote_b = 2;
   int64 total_voting_power = 3;
   int64 validator_power = 4;
-  Timestamp timestamp = 5 [(gogoproto.nullable) = false, (firehose.required) = true, (gogoproto.stdtime) = true];
+  Timestamp timestamp = 5
+      [(firehose.required) = true, (gogoproto.stdtime) = true];
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:99]: We don't need `firehose.required` here.

File: chain/cosmos/proto/type.proto [hunk 18]
```diff
@@ -97,16 +95,20 @@ message DuplicateVoteEvidence {
...
   int32 round = 3;
-  //BlockID block_id = 4 [(gogoproto.nullable) = false, (firehose.required) = true, (gogoproto.customname) = "BlockID"];
-  BlockId block_id = 4 [(gogoproto.nullable) = false, (firehose.required) = true, (gogoproto.customname) = "BlockID"];
-  Timestamp timestamp = 5 [(gogoproto.nullable) = false, (firehose.required) = true, (gogoproto.stdtime) = true];
+  // BlockID block_id = 4 [ (firehose.required) =
+  // true, (gogoproto.customname) = "BlockID"];
+  BlockId block_id = 4
+      [(firehose.required) = true, (gogoproto.customname) = "BlockID"];
+  Timestamp timestamp = 5
+      [(firehose.required) = true, (gogoproto.stdtime) = true];
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:111]: We don't need `firehose.required` here.

File: chain/cosmos/proto/type.proto [hunk 19]
```diff
@@ -97,16 +95,20 @@ message DuplicateVoteEvidence {
...
 
 message EventVote {
   SignedMsgType event_vote_type = 1 [json_name = "type"];
   uint64 height = 2;
   int32 round = 3;
-  //BlockID block_id = 4 [(gogoproto.nullable) = false, (firehose.required) = true, (gogoproto.customname) = "BlockID"];
-  BlockId block_id = 4 [(gogoproto.nullable) = false, (firehose.required) = true, (gogoproto.customname) = "BlockID"];
-  Timestamp timestamp = 5 [(gogoproto.nullable) = false, (firehose.required) = true, (gogoproto.stdtime) = true];
+  // BlockID block_id = 4 [ (firehose.required) =
+  // true, (gogoproto.customname) = "BlockID"];
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:107]: This comment looks like some kind of leftover. Could we remove it?

File: chain/cosmos/proto/type.proto [hunk 20]
```diff
@@ -116,18 +118,23 @@ enum SignedMsgType {
...
 }
 
 message LightClientAttackEvidence {
   LightBlock conflicting_block = 1;
   int64 common_height = 2;
   repeated Validator byzantine_validators = 3;
   int64 total_voting_power = 4;
-  Timestamp timestamp = 5 [(gogoproto.nullable) = false, (firehose.required) = true, (gogoproto.stdtime) = true];
+  Timestamp timestamp = 5
+      [(firehose.required) = true, (gogoproto.stdtime) = true];
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:137]: We don't need `firehose.required` here.

File: chain/cosmos/proto/type.proto [hunk 21]
```diff
@@ -143,25 +150,31 @@ message SignedHeader {
 message Commit {
   int64 height = 1;
   int32 round = 2;
-  //BlockID block_id = 3 [(gogoproto.nullable) = false, (firehose.required) = true, (gogoproto.customname) = "BlockID"];
-  BlockId block_id = 3 [(gogoproto.nullable) = false, (firehose.required) = true, (gogoproto.customname) = "BlockID"];
-  repeated CommitSig signatures = 4 [(gogoproto.nullable) = false, (firehose.required) = true];
+  // BlockID block_id = 3 [ (firehose.required) =
+  // true, (gogoproto.customname) = "BlockID"];
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:154]: Leftover comment?

File: chain/cosmos/proto/type.proto [hunk 22]
```diff
@@ -143,25 +150,31 @@ message SignedHeader {
...
+  BlockId block_id = 3
+      [(firehose.required) = true, (gogoproto.customname) = "BlockID"];
+  repeated CommitSig signatures = 4 [(firehose.required) = true];
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:157]: We don't need `firehose.required` here.

File: chain/cosmos/proto/type.proto [hunk 23]
```diff
@@ -143,25 +150,31 @@ message SignedHeader {
...
 }
 
 message CommitSig {
   BlockIDFlag block_id_flag = 1;
   bytes validator_address = 2;
-  Timestamp timestamp = 3 [(gogoproto.nullable) = false, (firehose.required) = true, (gogoproto.stdtime) = true];
+  Timestamp timestamp = 3
+      [(firehose.required) = true, (gogoproto.stdtime) = true];
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:164]: We don't need `firehose.required` here.

File: chain/cosmos/proto/type.proto [hunk 24]
```diff
@@ -173,28 +186,31 @@ message ValidatorSet {
 
 message Validator {
   bytes address = 1;
-  PublicKey pub_key = 2 [(gogoproto.nullable) = false, (firehose.required) = true];
+  PublicKey pub_key = 2 [(firehose.required) = true];
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:189]: We don't need `firehose.required` here.

File: chain/cosmos/proto/type.proto [hunk 25]
```diff
@@ -211,7 +227,7 @@ message ResponseEndBlock {
 
 message ValidatorUpdate {
   bytes address = 1;
-  PublicKey pub_key = 2 [(gogoproto.nullable) = false, (firehose.required) = true];
+  PublicKey pub_key = 2 [(firehose.required) = true];
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:230]: We don't need `firehose.required` here.

File: chain/cosmos/proto/type.proto [hunk 26]
```diff
@@ -229,7 +245,8 @@ message BlockParams {
 
 message EvidenceParams {
   int64 max_age_num_blocks = 1;
-  Duration max_age_duration = 2 [(gogoproto.nullable) = false, (firehose.required) = true, (gogoproto.stdduration) = true];
+  Duration max_age_duration = 2
+      [(firehose.required) = true, (gogoproto.stdduration) = true];
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:249]: We don't need `firehose.required` here.

File: chain/cosmos/proto/type.proto [hunk 27]
```diff
@@ -323,7 +337,10 @@ message CompactBitArray {
 }
 
 message Fee {
-  repeated Coin amount = 1 [(gogoproto.nullable) = false, (firehose.required) = true, (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"];
+  repeated Coin amount = 1 [
+    (firehose.required) = true,
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
+  ];
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:343]: We don't need `firehose.required` here.

File: chain/cosmos/proto/type.proto [hunk 28]
```diff
@@ -333,11 +350,15 @@ message Coin {
...
+  string amount = 2
+      [(gogoproto.customtype) = "Int", (firehose.required) = true];
 }
 
 message Tip {
-  repeated Coin amount = 1 [(gogoproto.nullable) = false, (firehose.required) = true, (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"];
+  repeated Coin amount = 1 [
+    (firehose.required) = true,
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
+  ];
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:361]: We don't need `firehose.required` here.

File: chain/cosmos/proto/type.proto [hunk 29]
```diff
@@ -348,7 +369,8 @@ message ResponseDeliverTx {
   string info = 4;
   int64 gas_wanted = 5;
   int64 gas_used = 6;
-  repeated Event events = 7 [(gogoproto.nullable) = false, (firehose.required) = true, (gogoproto.jsontag) = "events,omitempty"];
+  repeated Event events = 7
+      [(firehose.required) = true, (gogoproto.jsontag) = "events,omitempty"];
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:373]: We don't need `firehose.required` here.

File: chain/cosmos/proto/type.proto [hunk 30]
```diff
@@ -116,18 +118,23 @@ enum SignedMsgType {
...
-  SIGNED_MSG_TYPE_PRECOMMIT = 2 [(gogoproto.enumvalue_customname) = "PrecommitType"];
-  SIGNED_MSG_TYPE_PROPOSAL = 32 [(gogoproto.enumvalue_customname) = "ProposalType"];
+  SIGNED_MSG_TYPE_UNKNOWN = 0
+      [(gogoproto.enumvalue_customname) = "UnknownType"];
+  SIGNED_MSG_TYPE_PREVOTE = 1
+      [(gogoproto.enumvalue_customname) = "PrevoteType"];
+  SIGNED_MSG_TYPE_PRECOMMIT = 2
+      [(gogoproto.enumvalue_customname) = "PrecommitType"];
+  SIGNED_MSG_TYPE_PROPOSAL = 32
+      [(gogoproto.enumvalue_customname) = "ProposalType"];
```
Comment (pienkowb) [inline chain/cosmos/proto/type.proto:128]: This also looks like different formatting rules. Let's use this config: https://github.com/figment-networks/proto-cosmos/blob/main/.clang-format
